robotsim.py

```diff
@@ -166,6 +215,53 @@ def ultrasonicFront(self):
         clock.tick(120)
         return distance * 30
 
+    def scanEnvironment(self):
+        return map.tiles[self.row][self.col].envType
+
+    def detectFireFront(self):
+        # Map dir:
+        #   0 -> North
+        #   1 -> West
+        #   2 -> South
+        #   3 -> East
+        row = self.row
+        col = self.col
+        if self.dir == 0:
+            if row > 0 and map.tiles[row - 1][col].envType == "fire":
+                return True
+            return False
+        if self.dir == 1:
+            if col > 0 and map.tiles[row][col - 1].envType == "fire":
+                return True
+            return False
+        if self.dir == 2:
+            if row < map.height-1 and map.tiles[row + 1][col].envType == "fire":
+                return True
+            return False
+        if self.dir == 3:
+            if col < map.width-1 and map.tiles[row][col + 1].envType == "fire":
+                return True
+            return False
+        return False
```

Comment:
There is no need for a return False and return True after every nested if. You could just return the value of the condition. For example:
```bash
if self.dir == 0:
    return row > 0 and map.tiles[row - 1][col].envType == "fire"
```

Reply:
vava, ya acorté la función